trieste/models/keras/interface.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from abc import ABC, abstractproperty
+from typing import Optional
+
+import tensorflow as tf
+
+from ...types import TensorType
+from ..interfaces import ProbabilisticModel
+from ..optimizer import BatchOptimizer
+
+
+class KerasPredictor(ProbabilisticModel, tf.Module, ABC):
```

Comment:
high level question: why not inherit from `TrainableProbabilisticModel`?

Reply:
not sure honestly, with GPflux and here we followed the setup with GPflow - I think the idea is that these high level model interfaces are not meant to provide optimize and update methods, model wrappers are supposed to do that, so they inherit from both KerasPredictor and TrainableProbabilisticModel

perhaps @uri-granta can shed more light on the design here?